### ISSUE TITLE
Unknown arguments with dash are not recognized in strict mode

### ISSUE THREAD

**Issue by davazp** [yargs/yargs#1325]
When I provide an argument `--foo-baz`, it is not recognized as unknown in strict mode.

I modified a test to reproduce the problem:
https://github.com/yargs/yargs/compare/master...davazp:fix-strict-unknown-option-with-dash?expand=1

The problem is in `lib/validation.js` file, here:

https://github.com/yargs/yargs/blob/e7f29379707f9e3d5eb6edc09ba278f53cc7db74/lib/validation.js#L96-L99

the last part of the condition `!aliases.hasOwnProperty(key)` does not hold, because the argument `foo-baz` is aliased as `fooBaz` as well. As you can see here:

```
{ argv:
   { _: [],
     f: 10,
     foo: 10,
     bar: 20,
     b: 20,
     'foo-baz': 30,
     fooBaz: 30,
     '$0': './usage' },
  aliases:
   { help: [],
     version: [],
     foo: [ 'f' ],
     f: [ 'foo' ],
     bar: [ 'b' ],
     b: [ 'bar' ],
     'foo-baz': [ 'fooBaz' ],
     fooBaz: [ 'foo-baz' ] },
  positionalMap: {} }
```

Should `foo-bar` and `fooBaz` be in aliases at all, or only user-defined options should be in present there? 

If it is, then I guess we should not consider the alias for the camel case version.  For `fooBar`, `decamelize` could be used to check it but for `foo-bar` I couldn't find an easy way. I think this part is lives in `yarg-parser`?

Any hints about how do you prefer this to be fixed?

Thanks!

**Comment by juergba** [comment]
The default value of `camel-case-expansion` in `yargs-parser` is `true`. If you set this value to `false`, no alias is created and your argument `--foo-baz` will be recognized as unknown.
```js
var argv = yargs
    .usage('Usage: $0 -f NUM --bar NUM')
    .options({
        foo: { demand: 'foo option', alias: 'f' },
        bar: { demand: 'bar option', alias: 'b' }
    })
    .parserConfiguration({'camel-case-expansion': false})
    .strict()
    .parse('-f 10 --bar 20 --foo-Baz 30');
```
So I'm not sure wether this is really a bug or just an implication of the camel-case feature.

**Comment by juergba** [comment]
see also #1039
In that issue the behavior is classified as a bug.

**Comment by bcoe** [comment]
This should be addressed in `yargs@14.2.0` 🎉, please let us know if you continue to bump into any issues.
